src/sentry/lang/native/sources.py

```diff
@@ -149,6 +171,13 @@ def get_internal_source(project):
         ),
     )
 
+    if last_upload := get_last_upload(project.id):
```

Comment:
This endpoint uses `paginate()` and just wanted to double-check, that `reverse()` will give us just the path itself here. Not sure how `paginate()` works exactly, but if it adds some query to the path, simply joining two together might result in multiple `?` in the url.

Reply:
I don’t fully unterstand what you mean, what does this have to do with `paginate()`?

Reply:
`sentry_source_url` above (L166) is constructed with `reverse("sentry-api-0-dsym-files", {})`, which is effectively calling `DebugFilesEndpoint.get()`.
This method, returns the result of calling into `return self.paginate(...)`, which determines whether to append `per_page`, `cursor` and other pagination-related query strings.
Not sure how it exactly works, but my question was whether there's a chance that it will return a url path that already has _some_ query string attached to it. If so, then doing `sentry_source_url += f"?_last_upload={last_upload}"` may result in double `?` in the params.

Reply:
Or is pagination done entirely in the response header?

Reply:
Well pagination is done as a "response" at the very least. This is just contructing the url to query.

And symbolicator will then add its own query on to it, like the `debug_id` it is looking for.
The response of that *may* be paginated, but I believe that works using headers. Not entirely sure, I never dealt with pagination.